dotnet/src/webdriver/Internal/AsyncHelper.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="AsyncHelper.cs" company="WebDriver Committers">
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenQA.Selenium.Internal
+{
+    /// <summary>
+    /// Encapsulates methods for working with asynchronous tasks.
+    /// </summary>
+    public static class AsyncHelper
```

Comment:
The `AsyncHelper` implementation is similar to <https://github.com/aspnet/AspNetIdentity/blob/main/src/Microsoft.AspNet.Identity.Core/AsyncHelper.cs>. If you check the differences, this implementation lacks cultures set for async functions. Maybe worth adding as well, or cultures skipped intentionally for some reason? Also I would recommend making `AsyncHelper` class internal.

Reply:
We either need to duplicate the helper class in both webdriver and support packages or make it public in webdriver to access it from support. This approach appears to be what other classes in the repo do.

Reply:
Duplicate, it is not a criminal.

Reply:
Agreed, in this case I'm following what appears to be done elsewhere